### DISCUSSION TITLE
Extract `meteor build` to standalone NPM module

### DISCUSSION THREAD

**Discussion by maka-io** [meteor/meteor#11782] [Feature Requests]
During CI/CD pipeline builds, it's fairly common for building a meteor application is to:

1) Git clone repo into build context
2) Download and install meteor
3) Build application using "meteor build"
4) Save tarball as artifact to be handled by the deployment phase...

While the process of building the application is lengthy on its own, downloading the entire meteor distribution, and then extracting the 1GB package takes a significant amount of time as well.

If the command can be extracted out as a standalone npm module, the end to end time of this scenario's build / deploy would be much lower by doing something like `npm install -g @meteor/meteor-build`.

**Comment by filipenevola** [comment]
Hi, you should use docker layers in order to save time in your builds.

Each docker RUN is going to create a cache layer, you could use this image for example https://github.com/meteor/galaxy-images/tree/master/meteor-base

Also, everything that Meteor does is necessary in order to build the app, so moving to npm would require to move all the steps to npm as well.

If you have concrete steps that you think could be faster please describe them but just moving to npm is not going to change anything.
